### ISSUE TITLE
setAxisItems() + showGrid() makes zooming weird

### ISSUE THREAD

**Issue by temeddix** [pyqtgraph/pyqtgraph#2072]
### Short description
Using setAxisItems() with showGrid() method makes zooming(scrolling) weird.

### Code to reproduce
```
import sys
import math
from PyQt5 import QtWidgets
from pyqtgraph import PlotWidget, AxisItem


class CustomAxisItem(AxisItem):
    def __init__(self, *args, **kwargs):
        AxisItem.__init__(self, *args, **kwargs)

    def tickValues(self, minVal, maxVal, size):
        majticks = [math.sqrt(2) ** i for i in range(-20, 40)]
        majticks = [i for i in majticks if minVal < i < maxVal]
        return [(maxVal - minVal, majticks)]


class Window(QtWidgets.QMainWindow):
    def __init__(self):

        super().__init__()

        self.plot_widget = PlotWidget()
        axis_items = {
            "top": CustomAxisItem(orientation="top"),
            "bottom": CustomAxisItem(orientation="bottom"),
            "left": CustomAxisItem(orientation="left"),
            "right": CustomAxisItem(orientation="right"),
        }
        self.plot_widget.plotItem.setAxisItems(axis_items)
        self.plot_widget.plotItem.showGrid(x=True, y=True, alpha=0.2)
        self.setCentralWidget(self.plot_widget)

        self.line = self.plot_widget.plot(pen="#00AAFF")
        self.line.setData([0, 1, 2, 3, 4, 5], [0, 1, 4, 9, 16, 25])
        self.setCentralWidget(self.plot_widget)


if __name__ == "__main__":

    my_app = QtWidgets.QApplication(sys.argv)
    my_window = Window()
    my_window.showMaximized()
    my_app.exec()

```

### Expected behavior
Zooming to mouse position when scrolling.

### Real behavior
Zooming to weird position when scrolling.
No Excpetion occured.

https://user-images.githubusercontent.com/66480156/141398973-2df70332-53c9-4a26-a98e-55cbd566f15c.mp4

### Tested environment(s)

 * PyQtGraph version: 0.12.3
 * Qt Python binding: PyQt5 5.12.3 Qt 5.12.9
 * Python version: 3.8.12
 * NumPy version: 1.21.4
 * Operating system: Windows 11
 * Installation method: conda

### Additional context
Panning(MMB) still works well.

Removing either one of setAxisItem() or showGrid() makes this strange behavior disappear, like this:
```
        self.plot_widget = PlotWidget()
        self.plot_widget.plotItem.showGrid(x=True, y=True, alpha=0.2)
        self.setCentralWidget(self.plot_widget)
```
```
        self.plot_widget = PlotWidget()
        axis_items = {
            "top": CustomAxisItem(orientation="top"),
            "bottom": CustomAxisItem(orientation="bottom"),
            "left": CustomAxisItem(orientation="left"),
            "right": CustomAxisItem(orientation="right"),
        }
        self.plot_widget.plotItem.setAxisItems(axis_items)
        self.setCentralWidget(self.plot_widget)
```

Registering custom AxisItem from the widget creation also produces the same problem.
```
        axis_items = {
            "top": CustomAxisItem(orientation="top"),
            "bottom": CustomAxisItem(orientation="bottom"),
            "left": CustomAxisItem(orientation="left"),
            "right": CustomAxisItem(orientation="right"),
        }
        self.plot_widget = PlotWidget(axisItems=axis_items)
        self.plot_widget.plotItem.showGrid(x=True, y=True, alpha=0.2)
        self.setCentralWidget(self.plot_widget)
```

Using default AxisItem with showGrid() also produces the same problem.
```
        self.plot_widget = PlotWidget()
        axis_items = {
            "top": AxisItem(orientation="top"),
            "bottom": AxisItem(orientation="bottom"),
            "left": AxisItem(orientation="left"),
            "right": AxisItem(orientation="right"),
        }
        self.plot_widget.plotItem.setAxisItems(axis_items)
        self.plot_widget.plotItem.showGrid(x=True, y=True, alpha=0.2)
        self.setCentralWidget(self.plot_widget)
```

**Comment by temeddix** [comment]
This is the short video describing the problem:
https://user-images.githubusercontent.com/66480156/141398973-2df70332-53c9-4a26-a98e-55cbd566f15c.mp4

**Comment by j9ac9k** [comment]
I think we recently merged a fix for this in the master branch.  Can you
try installing from master and see if it is indeed fixed?

**Comment by temeddix** [comment]
I did a clean reinstall of pyqtgraph. Version is the latest 0.12.3, but the problem still exists.
![image](https://user-images.githubusercontent.com/66480156/141403850-6ed822ef-bc01-4f9f-a0c6-5125848d0426.png)

**Comment by j9ac9k** [comment]
Right; I'm saying i think a fix may have been merged after our most recent release, the way to test that is to run the example with pyqtgraph installed from the master branch.  I'll test out the minimum working example code you provided next time a hop on my machine 👍

**Comment by temeddix** [comment]
Oh, you meant this GitHub "master". I misunderstood for the latest release. Thank you for the quick reply. I'll try to apply the master code. Then would this fix be included in 0.12.4? If it will, when would it be?

**Comment by j9ac9k** [comment]
> Oh, you meant this GitHub "master". I misunderstood for the latest release. Thank you for the quick reply. Then would this fix be included in 0.12.4? If it will, when would it be?
> 
> P.S. I tried the master code from GitHub and it works well. It seems to be solved.

Glad to hear it's solved in master, I'll close this issue shortly.  

In terms of when 0.12.4 is released, not sure, soon; I was hoping it would be before this weekend, but this week has been a busy one for me after typical work hours.  Hopefully in the next week or two... Since 0.12.3 we've already had some critical bug fixes and a major performance improvement merged.
